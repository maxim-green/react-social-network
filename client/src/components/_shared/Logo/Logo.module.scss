.logo {
height: 40px;
  padding: 8px;
  a {
    display: block;
    height: 100%;
  }
  img {
    height: 100%;
  }
}