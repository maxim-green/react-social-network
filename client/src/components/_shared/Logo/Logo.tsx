import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from 'assets/images/logo.svg';
import classes from './Logo.module.scss';

export const Logo: React.FC = () => (
  <div className={classes.logo} data-testid="logo">
    <NavLink to="/">
      <img src={logo} alt="Bind" />
    </NavLink>
  </div>
);
