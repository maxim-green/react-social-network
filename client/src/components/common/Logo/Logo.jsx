import React from "react";
import {NavLink} from "react-router-dom";


const Logo = (props) => {
    return (
        <NavLink to='/'>
            <img src={props.src} alt="Bind"/>
        </NavLink>
    )
}

export default Logo