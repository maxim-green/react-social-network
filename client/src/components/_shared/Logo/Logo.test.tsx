import React from 'react';
import { screen, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Logo } from './Logo';

test('Component renders correctly', () => {
  render(<BrowserRouter><Logo /></BrowserRouter>);
  const logoComponent = screen.getByTestId('logo');
  expect(logoComponent).toBeInTheDocument();
  expect(logoComponent).toMatchSnapshot();
});
