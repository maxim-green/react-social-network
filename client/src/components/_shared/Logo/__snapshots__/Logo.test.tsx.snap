// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<div
  class="logo"
  data-testid="logo"
>
  <a
    aria-current="page"
    class="active"
    href="/"
  >
    <img
      alt="Bind"
      src="logo.svg"
    />
  </a>
</div>
`;
